vulnerabilities/importers/github.py

```diff
@@ -146,189 +122,339 @@
+                resp = requests.post(self.endpoint, headers=headers, json=query_json).json()
+                if resp.get("message") == "Bad credentials":
+                    raise GitHubTokenError("Invalid GitHub token")
+                advisories.extend(process_response(resp, ecosystem=ecosystem))
+                page_info = get_item(resp, "data", "securityVulnerabilities", "pageInfo")
+                if not get_item(page_info, "hasNextPage"):
+                    break
+                end_cursor = get_item(page_info, "endCursor")
+                if end_cursor:
+                    end_cursor = f'"{end_cursor}"'
+                    end_cursor_exp = f"after: {end_cursor}"
+        return advisories
+
+
+def get_ecosystem_from_type(type: str):
+    return PKG_TYPE_ECOSYSTEM_MAPPING.get(type)
+
+
+def get_type_from_ecosystem(ecosystem: str):
+    return ECOSYSTEM_PKG_TYPE_MAPPING.get(ecosystem)
+
+
+def extract_references(reference_data: List):
+    """
+    Yield `reference` by itearting over `reference_data`
+    """
+    for ref in reference_data:
+        url = ref["url"]
+        if not isinstance(url, str):
+            continue
+        if "GHSA-" in url.upper():
+            reference = Reference(url=url, reference_id=url.split("/")[-1])
+        else:
+            reference = Reference(url=url)
+        yield reference
+
+
+def process_name(ecosystem: str, pkg_name: str) -> Optional[Tuple[Optional[str], str]]:
+    """
+    Return `namespace` and `name` by processing `ecosystem` and `pkg_name`
+    """
+    if ecosystem == "MAVEN":
+        artifact_comps = pkg_name.split(":")
+        if len(artifact_comps) != 2:
+            return
+        ns, name = artifact_comps
+        return ns, name
+
+    if ecosystem == "COMPOSER":
         try:
-            self.gh_token = os.environ["GH_TOKEN"]
-        except KeyError:
-            raise GitHubTokenError("Environment variable GH_TOKEN is missing")
+            vendor, name = pkg_name.split("/")
+        except ValueError:
+            # TODO log this
+            return None
+        return vendor, name
 
-    def __enter__(self):
-        self.advisories = self.fetch()
+    if ecosystem in ("NUGET", "PIP", "RUBYGEMS", "GO"):
+        return None, pkg_name
 
-    def set_api(self, packages):
-        asyncio.run(self.version_api.load_api(packages))
 
-    def updated_advisories(self) -> Set[Advisory]:
-        return self.batch_advisories(self.process_response())
+class InvalidVersionRange(Exception):
+    pass
 
-    def fetch(self) -> Mapping[str, List[Mapping]]:
-        headers = {"Authorization": "token " + self.gh_token}
-        api_data = {}
-        for ecosystem in self.config.ecosystems:
 
-            api_data[ecosystem] = []
-            end_cursor_exp = ""
+def process_name_namespace(ecosystem: str, name: str, namespace: str) -> str:
+    """
+    Return `namespace` and `name` by processing `ecosystem` and `pkg_name`
```

Comment:
What's the use for this function? is this to recreate the naming convention that GitHub uses?

Reply:
Yes!